src/main/java/seedu/address/model/module/student/Student.java

```diff
@@ -22,19 +22,17 @@
     // Data fields
     private final StudentId studentId;
     //private final Set<Tag> tags = new HashSet<>();
-
     private UniqueTaskList taskList;
 
     /**
      * Every field must be present and not null.
      */
     public Student(StudentId studentId, Name name, TeleHandle teleHandle, Email email) {
-        requireAllNonNull(name, teleHandle, email, studentId);
+        requireAllNonNull(studentId, name, teleHandle, email);
+        this.studentId = studentId;
         this.name = name;
         this.teleHandle = teleHandle;
         this.email = email;
-        this.studentId = studentId;
-        this.taskList = new UniqueTaskList();
```

Comment:
I think u have omitted the task list initialization here so it becomes null and thus the test cases involving adding tasks to student won't work

Reply:
Thanks for the catch! Have updated to initialize student task list.